content/dev-tools/rpc-tool.page.tsx

```diff
@@ -0,0 +1,552 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import { useTranslate } from "@portal/hooks";
+import { useLocation } from "react-router-dom";
+import { Link } from "@redocly/portal/dist/client/App/Link";
+
+// Need this dynamic import for it to work with non-browser environments
+let ReactJson;
+if (typeof window !== 'undefined') {
+    ReactJson = require('react-json-view').default;
+}
+
+
+export default function RpcTool() {
+  const { hash: slug } = useLocation();
+  const [accountInfoResponse, setAccountInfoResponse] = useState(null);
+  const [accountLinesResponse, setAccountLinesResponse] = useState(null);
+  const [accountTxResponse, setAccountTxResponse] = useState(null);
+  const [accountObjectsResponse, setAccountObjectsResponse] = useState(null);
+  const [txResponse, setTxResponse] = useState(null);
+  const [ledgerResponse, setLedgerResponse] = useState(null);
+  const [inputText, setInputText] = useState(slug ? slug.slice(1) : "");
+  const [errorText, setErrorText] = useState(null);
+  const { translate } = useTranslate();
+  const [inputType, setInputType] = useState("");
+  const [showResult, setShowResult] = useState(false);
+  const [progressBarWidth, setProgressBarWidth] = useState("0%");
+  const [isAccountLinesExpanded, setIsAccountLinesExpanded] = useState(false);
+  const [isAccountTxExpanded, setIsAccountTxExpanded] = useState(false);
+  const [isAccountObjectsExpanded, setIsAccountObjectsExpanded] = useState(false);
+
+  const FULL_HISTORY_SERVER = "wss://s2.ripple.com";
+  const reTxId = /^[0-9A-Fa-f]{64}$/;
+  const reLedgerSeq = /^[0-9]+$/;
+
+  /*
+    XRPL requests
+    * account_info
+    * account_lines
+    * account_tx
+    * account_objects
+    * tx
+    * ledger
+  */
+
+  const accountInfo = async (
+    // @ts-expect-error - xrpl is added via a script tag
+    client: xrpl.Client,
+    address: string
+  ): Promise<void> => {
+    const response = await client.request({
+      command: "account_info",
+      account: address,
+    });
+    setProgressBarWidth("20%");
+    setAccountInfoResponse(response);
+  };
+
+  const accountLines = async (
+    // @ts-expect-error - xrpl is added via a script tag
```

Comment:
xrpl.js can now be directly imported - please do `npm i xrpl@3.0.0-beta.1` and then import normally :)